planner: supply description about plan cache supporting limit (#13090)
pingcap/docs: sql-prepared-plan-cache.md

- The query contains the `ignore_plan_cache` hint, such as `select /*+ ignore_plan_cache() */ * from t`.
- The query contains variables other than `?` (including system variables or user-defined variables), such as `select * from t where a>? and b>@x`.
- The query contains the functions that cannot be cached: `database()`, `current_user`, `current_role`, `user`, `connection_id`, `last_insert_id`, `row_count`, `version`, and `like`.

> - The query uses variables as the `LIMIT` parameter (such as `LIMIT ?` and `LIMIT 10, ?`) and the variable values are greater than 10000.

Author: ```suggestion
- The query uses a variable as the `LIMIT` parameter (such as `LIMIT ?` and `LIMIT 10, ?`) and the variable value is greater than 10000.
```
